dask/array/tests/test_cupy.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pytest
+
+import dask.array as da
+from dask.array.utils import assert_eq
+
+cupy = pytest.importorskip('cupy')
+
+
+functions = [
+    lambda x: x,
+    pytest.mark.xfail(lambda x: da.expm1(x), reason="cupy doesn't support expm1"),
+    lambda x: 2 * x,
+    lambda x: x / 2,
+    lambda x: x**2,
+    lambda x: x + x,
+    lambda x: x * x,
+    lambda x: x[0],
+    lambda x: x[:, 1],
+    lambda x: x[:1, None, 1:3],
+    lambda x: x.T,
+    lambda x: da.transpose(x, (1, 2, 0)),
+    lambda x: x.sum(),
+    pytest.mark.xfail(lambda x: x.dot(np.arange(x.shape[-1])), reason='cupy.dot(numpy) fails'),
+    pytest.mark.xfail(lambda x: x.dot(np.eye(x.shape[-1])), reason='cupy.dot(numpy) fails'),
+    pytest.mark.xfail(lambda x: da.tensordot(x, np.ones(x.shape[:2]), axes=[(0, 1), (0, 1)]),
+                      reason='cupy.dot(numpy) fails'),
+    lambda x: x.sum(axis=0),
+    lambda x: x.max(axis=0),
+    lambda x: x.sum(axis=(1, 2)),
+    lambda x: x.astype(np.complex128),
+    lambda x: x.map_blocks(lambda x: x * 2),
+    pytest.mark.xfail(lambda x: x.round(1), reason="cupy doesn't support round"),
```

Comment:
Could test `floor()`

Reply:
What is the motivation here?  I'm ok with marking things as xfail.  I don't necessarily want to find tests that will pass.  Copying over the sparse test suite for cupy was instructive to identify cases where the abstraction fails.  I'd like to keep them around.